src/common/ant.ts

```diff
@@ -93,11 +94,81 @@ export class AoANTReadable implements AoANTRead {
     }
 
     this.processId = this.process.processId;
+    this.checkHyperBeamPromise = this.checkHyperBeamCompatibility();
+  }
+
+  private async checkHyperBeamCompatibility(): Promise<boolean> {
+    const res = await fetch(
+      `https://permanode.xyz/${this.processId}~process@1.0/now/cache`,
+      {
+        method: 'HEAD',
+      },
+    );
+
+    if (res.ok && res.headers.get('initialized') === '"true"') {
+      return true;
+    }
+
+    return false;
   }
 
   async getState(
     { strict }: AntReadOptions = { strict: this.strict },
   ): Promise<AoANTState> {
+    // TODO: detect if process is hyperbeam compatible, if so fetch state via hyperbeam node
+    if (await this.checkHyperBeamPromise) {
+      const res = await fetch(
+        `https://permanode.xyz/${this.processId}~process@1.0/now/cache/serialize~json@1.0`,
+        {
+          method: 'GET',
+          redirect: 'follow',
+        },
+      );
+
+      if (!res.ok) {
+        throw new Error('Failed to fetch ant state');
+      }
+
+      const body = (await res.json()) as {
+        name: string;
+        ticker: string;
+        description: string;
+        keywords: string[];
+        denomination: string;
+        owner: string;
+        controllers: string[];
+        records: Record<string, { transactionid: string; ttlseconds: number }>;
+        balances: Record<string, number>;
+        logo: string;
+        totalsupply: number;
+        initialized: boolean;
+      };
+
+      return {
+        Name: body.name,
+        Ticker: body.ticker,
+        Description: body.description,
+        Keywords: body.keywords,
+        Denomination: parseInt(body.denomination),
+        Owner: body.owner,
+        Controllers: body.controllers,
+        Records: Object.entries(body.records).reduce(
+          (acc, [key, record]) => {
+            acc[key] = {
+              transactionId: record.transactionid,
+              ttlSeconds: record.ttlseconds,
+            };
+            return acc;
+          },
+          {} as Record<string, AoANTRecord>,
+        ),
+        Balances: body.balances,
```

Comment:
confirming with FWD that the json@1.0 device makes all keys lowercase and there is no way to avoid it - consider pulling this conversion to a utility function vs. baked into this implementation

Reply:
well thats mighty annoying. Is there a reason why it does that?